### ISSUE TITLE
Patrol reports test failed after second action

### ISSUE THREAD

**Issue by lewismainanala** [leancodepl/patrol#1429]
### Steps to reproduce

1. Setup according to https://patrol.leancode.co/getting-started/native
2. Run tests using patrol test --target integration_test/example_test.dart --flavor dev --verbose emulator-5554
3. Observe the test results/logs.

### Actual results

**Test file**
//import 'package:flutter_test/flutter_test.dart';
import 'package:nala/main_development.dart' as app;
import 'package:patrol/patrol.dart';

void main() {
  patrolTest(
    'Login tests',
    nativeAutomation: true,
    ($) async {
      app.main();

      await $('Log in').tap();
    },

https://github.com/leancodepl/patrol/assets/111293664/8b880e23-aadd-4ecd-b129-8c9c2f51ea8c

  );
}

Anytime I add a patrol command after  app.main(); say await $('Log in').tap(); , the test is always marked as failed also any command past await $('Log in').tap(); is never executed;

***MainActivityTest.java***

     package **.**;

import androidx.test.platform.app.InstrumentationRegistry;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameters;
import pl.leancode.patrol.PatrolJUnitRunner;

@RunWith(Parameterized.class)
public class MainActivityTest {
    @Parameters(name = "{0}")
    public static Object[] testCases() {
        PatrolJUnitRunner instrumentation = (PatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.setUp(MainActivity.class);
        instrumentation.waitForPatrolAppService();
        return instrumentation.listDartTests();
    }

    public MainActivityTest(String dartTestName) {
        this.dartTestName = dartTestName;
    }

    private final String dartTestName;

    @Test
    public void runDartTest() {
        PatrolJUnitRunner instrumentation = (PatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.runDartTest(dartTestName);
    }
}

***Gradle-App**

### Logs

        : 
        : > Task :app:connectedDevDebugAndroidTest
        : Starting 1 tests on Pixel_XL_API_31(AVD) - 12
        : 
        : #redacted.MainActivityTest > runDartTest[example_test][Pixel_XL_API_31(AVD) - 12] FAILED 
        :       java.lang.AssertionError: Dart test failed: example_test
        : 
        : Tests on Pixel_XL_API_31(AVD) - 12 failed: There was 1 failure(s).
        : 
        : > Task :app:connectedDevDebugAndroidTest FAILED

        FAILURE: Build failed with an exception.

        * What went wrong:
        Execution failed for task ':app:connectedDevDebugAndroidTest'.
        See the report at: file:///#redacted/build/app/reports/androidTests/connected/flavors/dev/index.html

        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.

        * Get more help at https://help.gradle.org

        BUILD FAILED in 37s
        : 
        : Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
        : 
        : You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
        : 
        : See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings
        : 
        : Execution optimizations have been disabled for 1 invalid unit(s) of work during this build to ensure correctness.
        : Please consult deprecation warnings for more details.
        : 865 actionable tasks: 8 executed, 857 up-to-date
✗ Failed to execute tests of apk with entrypoint test_bundle.dart on emulator-5554 (Gradle test execution failed with code 1) (37.9s)
Exception: Gradle test execution failed with code 1
#0      AndroidTestBackend.execute.<anonymous closure> (package:patrol_cli/src/android/android_test_backend.dart:140:9)
<asynchronous suspension>
#1      DisposeScope.run (package:dispose_scope/src/dispose_scope.dart:46:7)
<asynchronous suspension>
#2      AndroidTestBackend.execute (package:patrol_cli/src/android/android_test_backend.dart:106:5)
<asynchronous suspension>
#3      TestCommand._executorFor.<anonymous closure> (package:patrol_cli/src/commands/test.dart:279:9)
<asynchronous suspension>
#4      _NativeTestRunner.run.runTestsOnDevice (package:patrol_cli/src/test_runner.dart:245:15)
<asynchronous suspension>
#5      Future.wait.<anonymous closure> (dart:async/future.dart:522:21)
<asynchronous suspension>
#6      _NativeTestRunner.run (package:patrol_cli/src/test_runner.dart:260:5)
<asynchronous suspension>
#7      TestCommand.run (package:patrol_cli/src/commands/test.dart:194:21)
<asynchronous suspension>
#8      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#9      PatrolCommandRunner.runCommand (package:patrol_cli/src/runner/patrol_command_runner.dart:314:18)
<asynchronous suspension>
#10     PatrolCommandRunner.run (package:patrol_cli/src/runner/patrol_command_runner.dart:258:18)
<asynchronous suspension>
#11     patrolCommandRunner (package:patrol_cli/src/runner/patrol_command_runner.dart:73:20)
<asynchronous suspension>
#12     main (file:///Users/lewis/.pub-cache/hosted/pub.dev/patrol_cli-2.0.0/bin/main.dart:6:20)
<asynchronous suspension>

See the logs above to learn what happened. Also consider running with --verbose. If the logs still aren't useful, then it's a bug - please report it.
Uninstalling #redacted from sdk gphone64 arm64
Uninstalling#redacted.test from sdk gphone64 arm64
 FAIL  test_bundle.dart on emulator-5554

### Patrol version

patrol: 2.0.0
patrol_cli: 2.0.0

### Patrol Doctor output

You're using Patrol CLI 2.0, which has breaking changes.
Read the migration guide at https://patrol.leancode.co/v2.
Disable this warning by setting the PATROL_MIGRATED environment variable.

Patrol CLI version: 2.0.0
Program adb found in #redacted/Library/Android/sdk/platform-tools/adb
Env var $ANDROID_HOME set to #redacted/Library/Android/sdk
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller

### Flutter Doctor output

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.8, on macOS 13.4.1 22F82 darwin-arm64, locale
    en-KE)
[✓] Android toolchain - develop for Android devices (Android SDK version
    33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.79.2)
[✓] VS Code (version 1.73.1)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!

**Comment by lewismainanala** [comment]
any help looking at this @bartekpacia

**Comment by bartekpacia** [comment]
I think it's because you have an animation constantly spinning (even though it's not visible). Please read [this](https://patrol.leancode.co/finders/advanced#you-gotta-pump-it-up-but-which-one-to-use).

TL;DR Try using `SettlePolicy.trySettle`.

**Comment by lewismainanala** [comment]
Hey thanks @bartekpacia for pointing this out. _SettlePolicy.trySettle_ worked like a charm for me

**Comment by bartekpacia** [comment]
I suggest you set it globally, so you don't have to write it in every line:

```dart
patrolTest(
  'test description',
  config: PatrolTesterConfig(settlePolicy: SettlePolicy.trySettle),
  nativeAutomation: true,
  ($) async { /* your test */ },
);
```

**Comment by RuslikMer** [comment]
Hello, I have same issue
 > Task :app:connectedStagingDebugAndroidTest
        : Starting 1 tests on SM-S901E - 13
        : 
        : com.bambinifashion.bambinifashion.MainActivityTest > runDartTest[patrol_test][SM-S901E - 13] FAILED 
        :       java.lang.AssertionError: Dart test failed: patrol_test
        :       ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞═════════════════
        : Tests on SM-S901E - 13 failed: There was 1 failure(s).

added this to my test, but it's crashed same way
config: PatrolTesterConfig(settlePolicy: SettlePolicy.trySettle)

So I can't just open an app and wait for main page

**/patrol_test.dart**

void main() {
  patrolTest(
    'test scenarios',
    config: const PatrolTesterConfig(settlePolicy: SettlePolicy.trySettle),
    nativeAutomation: true,
    // nativeAutomatorConfig: const NativeAutomatorConfig(
    //   packageName: 'com.bambinifashion.bambinifashion'
    // ),
        ($) async {

      final base = Base(tester: $.tester);

      app.main();

      await waitUntilAppIsLoaded($.tester);

      // await base.menu.goToMenuCategory(base.menu.newIn);
      // await base.catalog.addToWishlist('guest');
      // await base.home.goToWishlist();
      // await base.wishlist.remove();
    },
  );
}


and this is what inside waitUntilAppIsLoaded method

Future<void> waitUntilAppIsLoaded(WidgetTester tester) async {
  final finder = find.byWidgetPredicate((widget) => widget is BannerTile);

  Log.it('Searching for BannerTile widget');

  await pumpUntilFound(
    tester,
    finder,
    timeout: const Duration(seconds: 20),
  );

  Log.it('BannerTile widget found');
}

and pumpUntilFound method

Future<void> pumpUntilFound(
    WidgetTester tester,
    Finder finder, {
      Duration timeout = const Duration(seconds: 10),
  }) async {
    bool timerDone = false;
    final timer = Timer(timeout, () => timerDone = true);
    while (timerDone != true) {
      await tester.pump(const Duration(milliseconds: 300));

      final found = tester.any(finder);
      if (found) {
        timerDone = true;
      }
    }

    timer.cancel();
  }

**Comment by RuslikMer** [comment]
and tried just run the app without any pump, same

void main() {
  patrolTest(
    'test scenarios',
    config: const PatrolTesterConfig(settlePolicy: SettlePolicy.trySettle),
    nativeAutomation: true,
    // nativeAutomatorConfig: const NativeAutomatorConfig(
    //   packageName: 'com.bambinifashion.bambinifashion'
    // ),
        ($) async {

      // final base = Base(tester: $.tester);

      app.main();

      // await waitUntilAppIsLoaded($.tester);

      // await base.menu.goToMenuCategory(base.menu.newIn);
      // await base.catalog.addToWishlist('guest');
      // await base.home.goToWishlist();
      // await base.wishlist.remove();
    },
  );
}

**Comment by RuslikMer** [comment]
and also in my case Patrol opening app twice, don't understand why it's happening
